### ISSUE TITLE
training

### ISSUE THREAD

**Issue by shmily-lxy** [YXB-NKU/Strip-R-CNN#11]
I'm currently training the DOTA dataset using the downloaded pre-trained weights, but the process seems to be stuck at the step of loading the weights. Below are some of my configurations and the console output content.
···
export CUDA_VISIBLE_DEVICES=0

WORK_DIR='xxx/work_dir/strip_rcnn/strip_rcnn_t_fpn_1x_dota_le90'
python tools/train.py configs/strip_rcnn/strip_rcnn_t_fpn_1x_dota_le90.py \
                    --work-dir ${WORK_DIR} \
                    --cfg-options seed=42
···
```python
backbone=dict(
        type='StripNet',
        embed_dims=[32, 64, 160, 256],
        k1s=[1,1,1,1],
        k2s=[19,19,19,19],
        drop_rate=0.1,
        drop_path_rate=0.15,
        depths=[3,3,5,2],
        init_cfg=dict(type='Pretrained', checkpoint="configs/strip_rcnn/checkpoints/stripnet_t.pth"),
        norm_cfg=dict(type='SyncBN', requires_grad=True)),
```
```
init cfg {'type': 'Pretrained', 'checkpoint': 'configs/strip_rcnn/checkpoints/stripnet_t.pth'}
2025-02-25 10:10:31,204 - mmcv - INFO - initialize StripNet with init_cfg {'type': 'Pretrained', 'checkpoint': 'configs/strip_rcnn/checkpoints/stripnet_t.pth'}
2025-02-25 10:10:31,204 - mmcv - INFO - load model from: configs/strip_rcnn/checkpoints/stripnet_t.pth
2025-02-25 10:10:31,205 - mmcv - INFO - load checkpoint from local path: configs/strip_rcnn/checkpoints/stripnet_t.pth
```

**Comment by YXB-NKU** [comment]
Thank you for your interest in our work!


First, I suggest you use the following code to check if the issue is with the weight file.

if `torch.load` can load it successfully, then you can try loading another model to see if it also gets stuck, such as the lsknet model. 

```python
import torch

# Check if the weight file can be loaded normally using torch.load
try:
    checkpoint_path = "configs/strip_rcnn/checkpoints/stripnet_t.pth"
    weights = torch.load(checkpoint_path, map_location='cpu')
    print("Weights loaded successfully with torch.load.")
except Exception as e:
    print(f"Failed to load weights: {e}")

# Try loading another model (e.g., oriented-rcnn) to see if the problem persists
try:
    another_model_path = "path_to_oriented_rcnn_checkpoint.pth"
    another_weights = torch.load(another_model_path, map_location='cpu')
    print("Another model weights loaded successfully.")
except Exception as e:
    print(f"Failed to load another model's weights: {e}")
```

**Comment by YXB-NKU** [comment]
Thank you for your interest in our work!


First, I suggest you use the following code to check if the issue is with the weight file.

if `torch.load` can load it successfully, then you can try loading another model to see if it also gets stuck, such as the lsknet model. 

```python
import os
import torch

# 检查文件路径和权限
checkpoint_path = "configs/strip_rcnn/checkpoints/stripnet_t.pth"
if not os.path.exists(checkpoint_path):
    print(f"Checkpoint file not found: {checkpoint_path}")
elif not os.access(checkpoint_path, os.R_OK):
    print(f"No read permission for checkpoint file: {checkpoint_path}")
else:
    print(f"Loading checkpoint from: {checkpoint_path}")

# 加载权重文件
try:
    checkpoint = torch.load(checkpoint_path, map_location='cpu')
    print("Checkpoint loaded successfully.")
except Exception as e:
    print(f"Error loading checkpoint: {e}")
```

**Comment by shmily-lxy** [comment]
Thank you. I have run the demo. It can be loaded successfully very quickly , but it took 20 minutes to load when I was running the training, and there was an error during the initialization process. 
```
2025-02-25 11:28:24,233 - mmcv - INFO - 
roi_head.bbox_head.fc_reg_theta.weight - torch.Size([1, 1024]): 
XavierInit: gain=1, distribution=normal, bias=0 
 
2025-02-25 11:28:24,233 - mmcv - INFO - 
roi_head.bbox_head.fc_reg_theta.bias - torch.Size([1]): 
XavierInit: gain=1, distribution=normal, bias=0 
 
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```

**Comment by YXB-NKU** [comment]
it's worth be noting that 
```python

fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).

```
1.this is not an error , it won't affect your training, so don't worry about it.

2. During the training, the 20 minutes is not loading model time, this is the system loading and processing the dataset.
